### DISCUSSION TITLE
Get the data asynchronously and do not update the data on the page

### DISCUSSION THREAD

**Discussion by wChenonly** [vuejs/pinia#1354] [Help and Questions]
### Reproduction

no

### Steps to reproduce the bug

The project initialization acquires data asynchronously, stores pinia, and then uses store data on the page, but the data is not updated.

For log view, you first go to the log of the page, followed by the initialized log.

In the picture, the outer layer shows that the array is empty, but there is actually data inside.

<img width="568" alt="image" src="https://user-images.githubusercontent.com/37658262/172769093-161e2481-89ae-4f03-9d25-c3271aa88e01.png">

### Expected behavior

I hope the page can use the data correctly.

### Actual behavior

The page does not have the correct responsive data

**Comment by wChenonly** [comment]
Does anyone know why?

**Reply by itsmnthn** [reply]
I'm facing similar issue. I'm doing direct mutation in `setTrades` (1. in image) action to `market.info.trades = [...[{obj1}, {obj2}]]` where `market` is simple object with `info` is class object and `[{obj1}, {obj2}]` is variable with array of 2 objects. and I'm doing deep copy to avoid unwanted mutation or side effects. it sets the state and when action is done it removes it automatically check 2. in Image and it mutates the length with 0 which I don't do check 3. in image.



![image](https://user-images.githubusercontent.com/25129674/174066086-f26796b7-2ddc-4061-94a1-5438af57e839.png)
